src/main.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import logging
+import ipdb
+import sys
+import torch
+import random
+import importlib
+import yaml
+from box import Box
+from pathlib import Path
+
+import src
+
+
+def main(args):
+    logging.info(f'Load the config from {args.config_path} and save it to {config.main.saved_dir}.')
+    config = Box.from_yaml(filename=args.config_path)
+    with open(Path(config.main.saved_dir) / 'config.yaml') as f:
+        yaml.dump(config, f, default_flow_style=False)
+
+    # Make the experiment results deterministic.
+    random.seed(config.main.random_seed)
+    torch.manual_seed(random.getstate()[1][1])
+    torch.cuda.manual_seed_all(random.getstate()[1][1])
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
+
+    logging.info('Create the device.')
+    if 'cuda' in config.trainer.device and not torch.cuda.is_available():
+        raise ValueError("The cuda is not available. Please set the device in the trainer section to 'cpu'.")
+    device = torch.device(config.trainer.device)
+
+    logging.info('Create the train and validation datasets.')
+    train_dataset = _get_instance(src.data.datasets, config.dataset, 'train')
+    valid_dataset = _get_instance(src.data.datasets, config.dataset, 'valid')
+
+    logging.info('Create the train and validation dataloaders.')
+    cls = getattr(src.data.datasets, config.dataset.name)
+    config.dataloader.kwargs.update(collate_fn=getattr(cls, 'collate_fn', None))
+    train_dataloader = _get_instance(src.data.dataloader, config.dataloader, train_dataset)
+    valid_dataloader = _get_instance(src.data.dataloader, config.dataloader, valid_dataset)
```

Comment:
The batch sizes of the training dataloader and the validation dataloader may be different?

Reply:
Yes, so we need kwargs `train_batch_size` and `valid_batch_size` in the dataloader section of the config.

Reply:
OK

Reply:
Have updated.